IDinsight/aaq_ud_template
# [Hotfix] cached refresh of rules

Reviewer: @lickem22 
Estimate: 30 min

-----


# [HOTFIX] Cached refresh of rules

## Ticket

No ticket

## Description, Motivation and Context

1. Applies the same fix as in core -- `/inbound/check` calls refresh unless there's cached rules from the same `RULE_REFRESH_FREQ` window
2. `RULE_REFRESH_FREQ` is configurable as environment variable
3. Added test cases
4. Removed dependency on private token for FAQT

## How Has This Been Tested?

1. Unit tests
2. Launched local containers to 
  1. check a request with `/inbound/check`, 
  2. add a new rule that'd match the request using the "Check new urgency rule" page
  3. make the same request in 1 multiple times in a row make sure we get the same result

## Checklist:

This checklist is a useful reminder of small things that can easily be forgotten.
Put an `x` in all the items that apply and remove any items that are not relevant to this PR.


[1]: http://chris.beams.io/posts/git-commit/
